packages/std/src/math/uint512.rs

```diff
@@ -0,0 +1,703 @@
+/// let a = Uint512::from(258u128);
+/// let b = Uint512::new([
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 0u8,
+///     0u8, 0u8, 0u8, 0u8, 0u8, 0u8, 1u8, 2u8,
+/// ]);
+/// assert_eq!(a, b);
+/// ```
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema)]
+pub struct Uint512(#[schemars(with = "String")] U512);
+
+impl Uint512 {
+    pub const MAX: Uint512 = Uint512(U512::MAX);
+
+    /// Creates a Uint512(value) from a big endian representation. It's just an alias for
+    /// `from_big_endian`.
+    pub fn new(value: [u8; 64]) -> Self {
+        Self::from_be_bytes(value)
+    }
+
+    /// Creates a Uint512(0)
+    pub const fn zero() -> Self {
+        Uint512(U512::zero())
+    }
+
+    pub fn from_be_bytes(value: [u8; 64]) -> Self {
+        Uint512(U512::from_big_endian(&value))
+    }
+
+    pub fn from_le_bytes(value: [u8; 64]) -> Self {
+        Uint512(U512::from_little_endian(&value))
+    }
+
+    /// Returns a copy of the number as big endian bytes.
+    pub fn to_be_bytes(self) -> [u8; 64] {
+        let mut result = [0u8; 64];
+        self.0.to_big_endian(&mut result);
+        result
+    }
+
+    /// Returns a copy of the number as little endian bytes.
+    pub fn to_le_bytes(self) -> [u8; 64] {
+        let mut result = [0u8; 64];
+        self.0.to_little_endian(&mut result);
+        result
+    }
+
+    pub fn is_zero(&self) -> bool {
+        self.0.is_zero()
+    }
+
+    pub fn checked_add(self, other: Self) -> Result<Self, OverflowError> {
+        self.0
+            .checked_add(other.0)
+            .map(Self)
+            .ok_or_else(|| OverflowError::new(OverflowOperation::Add, self, other))
+    }
+
+    pub fn checked_sub(self, other: Self) -> Result<Self, OverflowError> {
+        self.0
+            .checked_sub(other.0)
+            .map(Self)
+            .ok_or_else(|| OverflowError::new(OverflowOperation::Sub, self, other))
+    }
+
+    pub fn checked_mul(self, other: Self) -> Result<Self, OverflowError> {
+        self.0
+            .checked_mul(other.0)
+            .map(Self)
+            .ok_or_else(|| OverflowError::new(OverflowOperation::Mul, self, other))
+    }
+
+    pub fn checked_div(self, other: Self) -> Result<Self, DivideByZeroError> {
+        self.0
+            .checked_div(other.0)
+            .map(Self)
+            .ok_or_else(|| DivideByZeroError::new(self))
+    }
+
+    pub fn checked_rem(self, other: Self) -> Result<Self, DivideByZeroError> {
+        self.0
+            .checked_rem(other.0)
+            .map(Self)
+            .ok_or_else(|| DivideByZeroError::new(self))
+    }
+
+    pub fn checked_shr(self, other: u32) -> Result<Self, OverflowError> {
```

Comment:
The checked variants typically return `Option<Self>` (https://doc.rust-lang.org/stable/src/core/num/uint_macros.rs.html#720) instead of  `Result<Self>`.

Not a big deal, but something you might want to take into account, to make it consistent in the future.